server/sonar-server/src/main/java/org/sonar/server/platform/monitoring/CeDatabaseMonitor.java

```diff
@@ -35,13 +37,16 @@ public CeDatabaseMonitor(JmxConnectionFactory jmxConnectionFactory) {
 
   @Override
   public String name() {
-    return "Compute Engine Database";
+    return "Compute Engine Database Connection";
   }
 
   @Override
-  public LinkedHashMap<String, Object> attributes() {
+  public Optional<Map<String, Object>> attributes() {
     try (JmxConnection connection = jmxConnectionFactory.create(ProcessId.COMPUTE_ENGINE)) {
-      LinkedHashMap<String, Object> result = new LinkedHashMap<>();
+      if (connection == null) {
+        return Optional.absent();
+      }
+      Map<String, Object> result = new LinkedHashMap<>();
```

Comment:
minor memory usage optimization, but since we know the number of elements, we could create a map of 9 rather than 16 (default capacity)